### ISSUE TITLE
[mimir-distributed] Set readOnlyRootFilesystem to `true` for components which do not require a disk

### ISSUE THREAD

**Issue by mattmendick** [grafana/helm-charts#1401]
For components which do not require a disk, the mimir-distributed chart should set `readOnlyRootFilesystem` to `true`.

There may be more cases of this, but at least this should be the case for memcached instances created using the bitnami chart (specified here https://github.com/bitnami/charts/blob/master/bitnami/memcached/templates/deployment.yaml#L77). This should also be the case for the tokengen job as it does not require disk access either.

**Comment by 09jvilla** [comment]
@mattmendick assuming https://github.com/grafana/helm-charts/issues/1315 gets completed, is this issue still needed? 
I guess it still is for the tokengen job, as well as doing a quick pass to see if there are any other components which do not require a disk?

**Comment by mattmendick** [comment]
I think you're right, this one changes to become only the tokengen fix and a pass as you mention.

**Comment by 09jvilla** [comment]
@zanhsieh - if you completed this issue, can you please link the PR that closed it?

**Comment by zanhsieh** [comment]
@mattmendick 
You agree @09jvilla mentioned #1315 would get this issue resolved, or do you mean something else?

**Comment by 09jvilla** [comment]
No, the comment says that once 1315 gets resolved we still need to fix tokengen and do another review of the chart in case we missed anything. 

https://github.com/grafana/helm-charts/issues/1401#issuecomment-1154058793
